File: python/src/mpeg_o/tools/transport_decode_cli.py
"""Decode a .mots transport stream into a .mpgo file (v0.10 M70).

Parallel to Java {@code com.dtwthalion.mpgo.tools.TransportDecodeCli}
and ObjC {@code MpgoTransportDecode}.

Usage:
    python -m mpeg_o.tools.transport_decode_cli <input.mots> <output.mpgo>
"""
from __future__ import annotations

import argparse
import sys

from mpeg_o.transport.codec import transport_to_file


def main(argv: list[str] | None = None) -> int:
    parser = argparse.ArgumentParser(
        description="Decode an MPEG-O transport stream into a .mpgo file."
    )
    parser.add_argument("input", help="path to a .mots file")
    parser.add_argument("output", help="path to write the .mpgo file")
    args = parser.parse_args(argv)
    ds = transport_to_file(args.input, args.output)
    ds.close()
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: java/src/main/java/com/dtwthalion/mpgo/tools/TransportDecodeCli.java
/*
 * MPEG-O Java Implementation — v0.10 M70.
 * SPDX-License-Identifier: LGPL-3.0-or-later
 */
package com.dtwthalion.mpgo.tools;

import com.dtwthalion.mpgo.SpectralDataset;
import com.dtwthalion.mpgo.transport.TransportReader;

import java.nio.file.Path;

/**
 * Decode an MPEG-O transport stream into a .mpgo file. Parallel to
 * Python {@code mpeg_o.tools.transport_decode_cli} and ObjC
 * {@code MpgoTransportDecode}.
 *
 * <p>Usage:
 * <pre>
 *   java -cp target/classes:&lt;deps&gt; \
 *        com.dtwthalion.mpgo.tools.TransportDecodeCli \
 *        input.mots output.mpgo
 * </pre>
 */
public final class TransportDecodeCli {

    public static void main(String[] args) throws Exception {
        if (args.length < 2) {
            System.err.println("usage: TransportDecodeCli <input.mots> <output.mpgo>");
            System.exit(2);
        }
        try (TransportReader tr = new TransportReader(Path.of(args[0]));
             SpectralDataset ds = tr.materializeTo(args[1])) {
            // just materialize
        }
    }
}
